src/firefly/js/visualize/ui/FileUploadViewPanel.jsx

```diff
@@ -362,14 +362,14 @@ function getDetailsModel(tableModel, report, detailsTblId, UNKNOWN_FORMAT) {
     return details;
 }
 
-function TableDisplayOption({isMoc, summaryTblId, currentReport, currentSummaryModel}) {
+function TableDisplayOption({isMoc, summaryTblId, currentReport, currentSummaryModel, acceptMoc}) {
 
     const selectedTables = getFileFormat(currentReport) ?
         getSelectedRows('Table', summaryTblId, currentReport, currentSummaryModel) : [];
 
     if ( selectedTables.length < 1) return null;
 
-    if (isMoc) {
+    if (isMoc && !acceptMoc) {
```

Comment:
I am confused by `acceptMoc` flag.  What does it means?  Is it allow MOC file, or only take MOC file?
Also, there's a concept of `SUPPORTED_TYPES` in this panel, e.g.. REGION, Image, Table, etc.
Shouldn't MOC be added to them, instead of treated differently?
If `acceptMoc` is a way to define what this upload panel is allowed to accept, then should there be a more generic parameter, like `allowTypes` which takes an array of types it will allow.

Reply:
This is one of the problems that we have not yet gotten to with this component. It was planned with @kpuriIpac last PR but we put it off.

`SUPPORTED_TYPES` is barely used and it is wrong, since `SUPPORTED_TYPES` is really what the FileAnalyzer recognizes. For example, the FileAnalyzer does not recognize spectrums.

The upload panel additionally recognizes the following tables:  
- MOC
- spectrum tables 
- Datalink tables (and I am adding in another branch) 


So we do need a list of what is allowed.

- region
- any table
- spectrum tables
- moc tables
- service descriptor tables (in another PR)
- images
- VO UWS xml file (future)

It seems to me this needs to be a stand alone task. I have not yet made a ticket for it so I probably should.